app/src/main/java/com/willowtree/vocable/presets/AddToCategoryPickerFragment.kt

```diff
@@ -0,0 +1,162 @@
+import com.willowtree.vocable.utils.VocableFragmentStateAdapter
+import kotlin.math.min
+
+class AddToCategoryPickerFragment : BaseFragment<FragmentAddToCategoryPickerBinding>() {
+
+    private val args: AddToCategoryPickerFragmentArgs by navArgs()
+
+    override val bindingInflater: BindingInflater<FragmentAddToCategoryPickerBinding> =
+        FragmentAddToCategoryPickerBinding::inflate
+    private lateinit var addToCategoryPickerViewModel: AddToCategoryPickerViewModel
+
+    private var maxCategories = 1
+    private lateinit var categoriesAdapter: CategoriesPagerAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val phraseText = args.phraseText
+
+        binding.backButton.action = {
+            findNavController().popBackStack()
+        }
+
+        val numColumns = resources.getInteger(R.integer.custom_category_columns)
+        val numRows = resources.getInteger(R.integer.custom_category_rows)
+        maxCategories = numColumns * numRows
+
+        categoriesAdapter = CategoriesPagerAdapter(childFragmentManager)
+
+        binding.categoryPagerForwardButton.action = {
+            when (val currentPosition = binding.categoryHolder.currentItem) {
+                categoriesAdapter.itemCount - 1 -> {
+                    binding.categoryHolder.setCurrentItem(0, true)
+                }
+                else -> {
+                    binding.categoryHolder.setCurrentItem(currentPosition + 1, true)
+                }
+            }
+        }
+
+        binding.categoryPagerBackButton.action = {
+            when (val currentPosition = binding.categoryHolder.currentItem) {
+                0 -> {
+                    binding.categoryHolder.setCurrentItem(
+                        categoriesAdapter.itemCount - 1,
+                        true
+                    )
+                }
+                else -> {
+                    binding.categoryHolder.setCurrentItem(currentPosition - 1, true)
+                }
+            }
+        }
+
+        binding.categoryHolder.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                val pageNum = position % categoriesAdapter.numPages + 1
+                binding.categoryPageNumber.text = getString(R.string.phrases_page_number, pageNum, categoriesAdapter.numPages)
+
+                val activity = requireActivity()
+                if (activity is MainActivity) {
+                    activity.resetAllViews()
+                }
+            }
+        })
+
+        addToCategoryPickerViewModel = ViewModelProviders.of(
+            requireActivity(),
+            BaseViewModelFactory()
+        ).get(AddToCategoryPickerViewModel::class.java)
+
+        subscribeToViewModel()
+
+        with(addToCategoryPickerViewModel) {
+            buildCategoryList(phraseText)
+        }
```

Comment:
Is there a benefit or point to using the "with" here?

Reply:
Lol nope.  I'll fix it